csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -192,6 +223,31 @@ public static string ToDiagnosticString(IMessage message)
             return diagnosticFormatter.Format(message);
         }
 
+        /// <summary>
+        /// Converts the field of a message to JSON for diagnostic purposes with no extra context.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This differs from calling <see cref="Format(IMessage, FieldDescriptor)"/> on the default JSON
+        /// formatter in its handling of <see cref="Any"/>. As no type registry is available
+        /// in <see cref="object.ToString"/> calls, the normal way of resolving the type of
+        /// an <c>Any</c> message cannot be applied. Instead, a JSON property named <c>@value</c>
+        /// is included with the base64 data from the <see cref="Any.Value"/> property of the message.
+        /// </para>
+        /// <para>The value returned by this method is only designed to be used for diagnostic
+        /// purposes. It may not be parsable by <see cref="JsonParser"/>, and may not be parsable
+        /// by other Protocol Buffer implementations.</para>
+        /// </remarks>
+        /// <param name="message">The message to format for diagnostic purposes.</param>
+        /// <param name="field">The field in the message to format for diagnostic purposes.</param>
+        /// <returns>The diagnostic-only JSON representation of the message</returns>
+        public static string ToDiagnosticString(IMessage message, FieldDescriptor field)
```

Comment:
How would you feel about us exposing the diagnostic formatter instead? I think this is going to be called relatively rarely compared with `ToDiagnosticString(message)` (which I can imagine using from a watch for example)...